src/librustdoc/html/render/mod.rs

```diff
@@ -3517,14 +3520,18 @@ fn render_assoc_items(
                 RenderMode::Normal
             }
             AssocItemRender::DerefFor { trait_, type_, deref_mut_ } => {
+                let id =
```

Comment:
It's still possible to have ID duplicates in your deref. Could you add a check over `id_map` too? The idea of `id_map` is to append a `-[NUMBER]` in case this ID already exists. Basically, I think it should look like this:

```rust
let id = cx.derive_id(small_url_encode(&format!("deref-methods-{:#}", type_.print())));
let id = cx.id_map.derive(id);
```

Don't know what you think about it @jyn514 ?

Reply:
Hmm, wait, isn't this already happening? `cx.derive_id` looks like this:

```rust
    fn derive_id(&self, id: String) -> String {
        let mut map = self.id_map.borrow_mut();
        map.derive(id)
    }
```

...so I believe this is already happening.

Reply:
Outch sorry, I didn't pay attention enough. You're absolutely right!